### PR TITLE
basic implementation on how to connect to firebase

### PR DESCRIPTION
**Pull request by sebastian-vangrieken**
The basic infrastructure to use firebase is present. We still have to decide how our folder structure for data access will look so I have only implemented a function to get all users from firebase. I will create the rest of the functionality in a separate branch with the folder structure we decided on

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -5,5 +5,7 @@ plugins {
     alias(libs.plugins.androidLibrary) apply false
     alias(libs.plugins.jetbrainsCompose) apply false
     alias(libs.plugins.kotlinMultiplatform) apply false
+
     alias(libs.plugins.sqlDelight) apply false
+    id("com.google.gms.google-services").version("4.4.1").apply(false)
```
Comment (DanneD-Wisemen) [inline build.gradle.kts]: Try using the version catalog

File: composeApp/build.gradle.kts [hunk 2]
```diff
@@ -4,6 +4,8 @@ plugins {
     alias(libs.plugins.kotlinMultiplatform)
     alias(libs.plugins.androidApplication)
     alias(libs.plugins.jetbrainsCompose)
+
+    id("com.google.gms.google-services") version "4.4.1"
```
Comment (DanneD-Wisemen) [inline composeApp/build.gradle.kts]: Try using the version catalog

File: composeApp/build.gradle.kts [hunk 3]
```diff
@@ -24,6 +26,8 @@ kotlin {
             implementation(libs.kotlinx.coroutines.android)
             implementation(libs.koinAndroid)
             implementation(libs.koinCore)
+
+            implementation(project.dependencies.platform("com.google.firebase:firebase-bom:32.7.4"))
```
Comment (DanneD-Wisemen) [inline composeApp/build.gradle.kts]: Try using the version catalog

File: composeApp/src/androidMain/kotlin/App.kt [hunk 4]
```diff
@@ -5,22 +5,26 @@ import androidx.compose.foundation.layout.fillMaxWidth
...
 import org.jetbrains.compose.ui.tooling.preview.Preview
 
-import chargehub.composeapp.generated.resources.Res
-import chargehub.composeapp.generated.resources.compose_multiplatform
-
 @OptIn(ExperimentalResourceApi::class)
 @Composable
 @Preview
-fun App(firstUser: String? = null) {
+fun App(firstUser: String? = null){
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/App.kt]: the space before the accolade is a default code style

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/MainActivity.kt [hunk 5]
```diff
@@ -2,17 +2,14 @@ package com.wisemen.chargehub
 
 import App
 import android.os.Bundle
-import android.util.Log
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/MainActivity.kt]: Good cleaning of the imports!!

File: iosApp/iosApp/iOSApp.swift [hunk 6]
```diff
@@ -1,11 +1,13 @@
 import SwiftUI
 import Shared
+import Firebase
 
 @main
 struct iOSApp: App {
 
 	init() {
-	   HelperKt.doInitKoin()
+        FirebaseApp.configure()
```
Comment (DanneD-Wisemen) [inline iosApp/iosApp/iOSApp.swift:9]: Check the indentation

File: shared/src/commonMain/kotlin/db/repository/user/RemoteFirebaseRepository.kt [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+import db.database.firebase.User
+import dev.gitlive.firebase.Firebase
+import dev.gitlive.firebase.firestore.firestore
+import org.koin.core.component.KoinComponent
+
+class RemoteFirebaseRepository: FirebaseRepository, KoinComponent {
+    private val firestore = Firebase.firestore
+
+    override suspend fun getUsers(): List<User> {
```
Comment (DanneD-Wisemen) [inline shared/src/commonMain/kotlin/db/repository/user/RemoteFirebaseRepository.kt]: You dont need a FirebaseRepository, this call should be in the UserRepository

File: shared/src/commonMain/kotlin/di/Helper.kt [hunk 8]
```diff
@@ -3,9 +3,16 @@ package di
 import org.koin.core.context.startKoin
 import org.koin.dsl.KoinAppDeclaration
 
+private val allModules = listOf(modules, clientsModule, servicesModule, repositoriesModule, platformModules())
```
Comment (DanneD-Wisemen) [inline shared/src/commonMain/kotlin/di/Helper.kt]: I do not see the added value of extracting this to another value 😅

**Review by DanneD-Wisemen — CHANGES_REQUESTED** [review]

File: shared/src/commonMain/kotlin/di/Helper.kt (hunk 8 above)
Comment (sebastian-vangrieken) [reply shared/src/commonMain/kotlin/di/Helper.kt]: Yarin told me to do it like this.

Comment (sebastian-vangrieken) [reply shared/src/commonMain/kotlin/di/Helper.kt]: The way to initialize koin is also slightly different for android and ios so we extracted it into a variable so we don't have to type it twice.

Comment (DanneD-Wisemen) [reply shared/src/commonMain/kotlin/di/Helper.kt]: Okay nice!

**Review by DanneD-Wisemen — APPROVED** [review]

**Pull request merged**